app/audit/models/CreateIncomeSourceAuditModel.scala

```diff
@@ -0,0 +1,92 @@
+ */
+
+package audit.models
+
+import audit.Utilities.{userAuditDetails, userAuditDetailsNino}
+import auth.MtdItUser
+import enums.IncomeSourceJourney.{IncomeSourceType, SelfEmployment}
+import models.createIncomeSource.CreateIncomeSourceResponse
+import models.incomeSourceDetails.viewmodels.{CheckBusinessDetailsViewModel, CheckDetailsViewModel, CheckPropertyViewModel}
+import play.api.libs.json.{JsObject, JsValue, Json}
+import utils.Utilities._
+
+
+case class CreateIncomeSourceAuditModel(incomeSourceType: IncomeSourceType,
+                                        viewModel: CheckDetailsViewModel,
+                                        failureCategory: Option[String],
+                                        failureReason: Option[String],
+                                        createIncomeSourceResponse: Option[CreateIncomeSourceResponse]
+                                       )(implicit user: MtdItUser[_]) extends ExtendedAuditModel {
+
+
+  private val isSuccessful = failureCategory.isEmpty
+  override val transactionName: String = enums.TransactionName.CreateIncomeSource
+  override val auditType: String = enums.AuditType.CreateIncomeSource
+
+  private val outcome: JsObject = {
+    val outcome: JsObject = Json.obj("isSuccessful" -> isSuccessful)
+
+    if (isSuccessful) outcome
+    else outcome ++ Json.obj(
+      "isSuccessful" -> isSuccessful,
+      "failureCategory" -> failureCategory,
+      "failureReason" -> failureReason)
+  }
+
+
+  override val detail: JsValue = {
+
+    val baseDetails = userAuditDetailsNino(user) ++
+      Json.obj(
+        "outcome" -> outcome,
+        "journeyType" -> incomeSourceType.journeyType
+      ) ++
+      ("addedIncomeSourceID", createIncomeSourceResponse.map(x => x.incomeSourceId))
+
+    val businessDetails = viewModel match {
+      case businessDetailsViewModel: CheckBusinessDetailsViewModel =>
+        val seDetails = Json.obj() ++
+          ("businessName", Some(businessDetailsViewModel.businessName)) ++
+          ("dateStarted", Some(businessDetailsViewModel.businessStartDate)) ++
+          Json.obj("businessDescription" -> businessDetailsViewModel.businessTrade) ++
+          Json.obj("addressLine1" -> businessDetailsViewModel.businessAddressLine1) ++
+          ("addressLine2", businessDetailsViewModel.businessAddressLine2) ++
+          ("addressLine3", businessDetailsViewModel.businessAddressLine3) ++
+          ("addressTownOrCity", businessDetailsViewModel.businessAddressLine4) ++
+          ("addressPostcode", businessDetailsViewModel.businessPostalCode) ++
+          ("addressCountry", businessDetailsViewModel.businessCountryCode)
+
+        seDetails
+
+      case propertyViewModel: CheckPropertyViewModel =>
+        val propDetails = Json.obj("dateStarted" -> propertyViewModel.tradingStartDate)
+
+        propDetails
+    }
+
+    val accountingMethod = if (viewModel.cashOrAccruals.toLowerCase.equals("cash")) {
+      "Cash basis accounting"
```

Comment:
instead of "cash basis accounting" - can we preserve the API vals eg CASH or ACCRUALS?

Reply:
We can however the example json from ticket does have -   "accountingMethod":"cash basis accounting"   
Are we better keeping the vals?

Reply:
resolved in recent commit